pkg/lint/rules/values.go

```diff
@@ -63,5 +63,5 @@ func validateValuesFile(valuesPath string) error {
 	if err != nil {
 		return err
 	}
-	return chartutil.ValidateAgainstSingleSchema(values, schema)
+	return chartutil.ValidateAgainstSingleSchema(chartutil.CoalesceTables(values, fv), schema)
```

Comment:
I took a bit of inspiration from https://github.com/helm/helm/blob/v3.0.2/pkg/lint/rules/template.go#L64. That being said, I noticed some potential oddities when poking around https://github.com/helm/helm/blob/v3.0.2/pkg/chartutil/coalesce.go. 

First, as per the first link, template linting invokes `chartutil.CoalesceValues(chrt *chart.Chart, vals map[string]interface{}) (Values, error)`. Somewhere along the line, this in turn calls `coalesceValues(c *chart.Chart, v map[string]interface{})` which handles special cases like removing keys for null values. The function I'm invoking here, on the other hand, doesn't contain any code paths that invoke that same logic. Is that going to be a problem?

Second, `chartutil.CoalesceValues(chrt *chart.Chart, vals map[string]interface{}) (Values, error)` operates on a `*chart.Chart` object. This means that it properly handles values from chart dependencies, etc. Per https://helm.sh/docs/topics/charts/#schema-files, the final `.Values` object is checked against all subchart schemas. Does this mean that `values.yaml` linting needs to respect more than just the parent chart's `values.yaml` file and JSON schema? If so, I think the changes will likely be much more involved than what I've done here

Reply:
Also, I wasn't sure what to do in terms of testing. Currently, there isn't really any testing around `values.yaml` linting. I could certainly add some unit tests for this function, but they wouldn't have caught this issue. Catching this issue would've likely required some higher-level integration tests that actually execute the CLI. Do you all think that this change requires additional test code? If so, what do you all recommend?